lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -2875,4 +2875,64 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
                 /*Torch_BoolType:$antialias*/ cstFalse);
         return success();
       });
+  patterns.onOp(
+      "Upsample", 9, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        std::string mode;
+        Value input, scales;
+        if (binder.tensorOperands(input, scales) ||
```

Comment:
You don't seem to be using the scales. 

At least for bilinear upsampling, the scale information is important to compute the weights in the average over nearby pixels. For nearest mode, do you know if they use round prefer floor (in onnx and/or pytorch), or just floor to determine nearest pixel? If it's round prefer floor, the scales might still be relevant. 

Also, if it doesn't matter in the end, would it be better to write this conversion by just passing the scales to the aten.upsample_nearest2d op instead of getting the sizes? That's not a review requirement, but an honest question. 

Either way, I feel like it should be documented in a comment how this op does rounding. These resizing ops are too varied in their implementations between each framework, so any commentary you can add would be really valuable to others who need to add functionality (like different modes) in the future.